packages/cf-style-container/src/createComponent.js

```diff
@@ -0,0 +1,72 @@
+import { extractPassThroughProps, resolvePassThrough } from 'fela-utils';
+import { createElement } from 'react';
+import PropTypes from 'prop-types';
+
+function createComponentFactory(createElement, contextTypes) {
+  return function createComponent(rule, type, passThroughProps) {
+    const displayName = rule.name ? rule.name : 'FelaComponent';
+
+    const FelaComponent = ({ children, ...ruleProps }, { renderer, theme }) => {
+      if (!renderer) {
+        throw new Error(
+          "createComponent() can't render styles without the renderer in the context. Missing react-fela's <Provider /> at the app root?"
```

Comment:
Those are two different things. `createComponent` should crash if there's missing `renderer` in the context.

The other bug is because our components have hard assumption that our base theme (cf-style-const) is in the context (ThemeProvider puts it into that context or in our case StyleProvider). I used to have a warning for that but it had to be in every component, which was annoying. **Now, I just put it here as an another error message.**